optimizer-fix-controls.md

```diff
@@ -26,6 +26,12 @@ SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
 
 ## Optimizer Fix Controls reference
 
+### [`33031`](https://github.com/pingcap/tidb/issues/33031) <span class="version-mark">New in v8.0.0</span>
+
+- Default value: `OFF`
```

Comment:
@mjonss  The idea of fix control is to maintain the behaviors which could cause plan change.  I don't think it's applicable here.   If you think it's necessary to control it, you may include a variable.

Reply:
@songrijie , it was suggested by @qw4990 and I agree that it may be a good way to allow disabling plan cache for partitioned tables, if there are bad cases of cached plans, like where a change in the queries partition pruning would affect the plan/execution which the cached plan would behave differently. I don't see it possible, but this may help to work around possible bugs.

So I do think it is related to other fix controls, since it may affect plan changes (i.e. by allowing plans to be cached, which they were not before this change).

Reply:
@mjonss  ok, in this case, is it appropriate to make it `ON` by default?   As it's not easy for users to acknowledge this control, we shall make the default value suitable for most scenarios.  

Please mention this fix control in the doc of plan cache as well.

Reply:
@songrijie I would still have this as `OFF` by default, to allow partitioned tables to make use of plan cache, when it is using the default `tidb_partition_prune_mode = 'dynamic'`. If set to `ON` it will not allow any plans from the plan cache for partitioned tables. The TiDB PR https://github.com/pingcap/tidb/pull/49161 is to enable plan cache for partitioned tables, and this fix control setting is to make it possible to block the plan cache for partitioned tables.

Reply:
@mjonss oh, sorry, I misunderstand the value. I meant to enable it  by default.

Reply:
@songrijie I would compare this fix control to https://docs.pingcap.com/tidb/stable/optimizer-fix-controls#44830-new-in-v657-and-v730

Reply:
LGTM

Reply:
@songrijie since fix-control 44830 is not mentioned anywhere else, should I still add 33031 in both the `SQL Non-Preparted Execution Plan Cache` and `SQL Prepared Execution Plan Cache`?

Reply:
@mjonss it's not necessary for now. As long as we have rare case to disable it, we can leave it for internal use.